src/main/java/com/nha/abdm/wrapper/hip/hrp/link/userInitiated/LinkService.java

```diff
@@ -86,14 +99,28 @@ public void onInit(InitResponse initResponse) {
 
     log.info("onInit body : " + onInitRequest.toString());
     try {
-      ResponseEntity<GenericResponse> responseEntity =
-          requestManager.fetchResponseFromGateway(onInitLinkPath, onInitRequest);
-      log.info(onInitLinkPath + " : onInitCall: " + responseEntity.getStatusCode());
+      log.info("Sending otp request to HIP");
+      RequestOtp requestOtp = RequestOtp.builder().abhaAddress(initResponse.getPatient().getId()).patientReference(initResponse.getPatient().getReferenceNumber()).build();
+      ResponseEntity<ResponseOtp> hipResponse=hipClient.fetchResponseFromHIPForOtp(this.requestOtp, requestOtp);
+      log.info(this.requestOtp + " : requestOtp: " + hipResponse.getStatusCode());
+
+      if(Objects.requireNonNull(hipResponse.getBody()).getError()==null){
```

Comment:
lets again set the contract right. HIP should send an acknowledgement. lets verify on that instead of checking on error